pybamm/solvers/c_solvers/idaklu/CasadiSolverOpenMP.cpp

```diff
@@ -0,0 +1,550 @@
+#include "CasadiSolverOpenMP.hpp"
+#include "casadi_sundials_functions.hpp"
+#include <casadi/casadi.hpp>
+#include <casadi/core/function.hpp>
+#include <casadi/core/sparsity.hpp>
+
+/*
+ * This is an abstract class that implements an OpenMP solution but
```

Comment:
can we please move the documentation into the header files please? Ideally in doxygen format, but that is not neccessary at this point since we don't generate any api docs

Reply:
👍